src/controllers/main/main.ts

```diff
@@ -1602,41 +1616,38 @@ export class MainController extends EventEmitter {
           nonce: Number(accountOp.nonce)
         }
       } catch (e: any) {
-        return this.#throwAccountOpBroadcastError(e)
+        return this.#throwBroadcastAccountOp(e)
       }
     }
 
-    if (transactionRes) {
-      const submittedAccountOp: SubmittedAccountOp = {
-        ...accountOp,
-        status: AccountOpStatus.BroadcastedButNotConfirmed,
-        txnId: transactionRes.hash,
-        nonce: BigInt(transactionRes.nonce),
-        timestamp: new Date().getTime(),
-        isSingletonDeploy: !!accountOp.calls.find((call) => getAddress(call.to) === SINGLETON)
-      }
-      if (accountOp.gasFeePayment?.isERC4337) {
-        submittedAccountOp.userOpHash = transactionRes.hash
-      }
-      await this.activity.addAccountOp(submittedAccountOp)
-      await this.resolveAccountOpAction(
-        {
-          hash: transactionRes?.hash || null,
-          networkId: network.id,
-          isUserOp: !!accountOp?.asUserOperation
-        },
-        actionId
+    if (!transactionRes)
+      return this.#throwBroadcastAccountOp(
+        new Error('No transaction response received after being broadcasted.')
       )
 
-      console.log('broadcasted:', transactionRes)
-      this.onSignSuccess('account-op')
-      this.broadcastStatus = 'DONE'
-      this.emitUpdate()
-      await wait(1)
+    const submittedAccountOp: SubmittedAccountOp = {
+      ...accountOp,
+      status: AccountOpStatus.BroadcastedButNotConfirmed,
+      txnId: transactionRes.hash,
+      nonce: BigInt(transactionRes.nonce),
+      timestamp: new Date().getTime(),
+      isSingletonDeploy: !!accountOp.calls.find((call) => getAddress(call.to) === SINGLETON)
     }
+    if (accountOp.gasFeePayment?.isERC4337) {
+      submittedAccountOp.userOpHash = transactionRes.hash
+    }
+    await this.activity.addAccountOp(submittedAccountOp)
+    await this.resolveAccountOpAction(
+      {
+        hash: transactionRes?.hash || null,
+        networkId: network.id,
+        isUserOp: !!accountOp?.asUserOperation
+      },
+      actionId
+    )
 
-    this.broadcastStatus = 'INITIAL'
-    this.emitUpdate()
+    this.onSignSuccess('account-op')
+    return Promise.resolve(submittedAccountOp)
```

Comment:
Is this needed?

Reply:
Otherwise, TypeScript complains that not all code paths return a value. I could have returned void (`Promise.resolve()`), but I returned the submitted account op instead.